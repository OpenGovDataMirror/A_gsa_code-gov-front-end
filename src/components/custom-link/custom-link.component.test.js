import React from 'react'
import { shallow } from 'enzyme'

import CustomLink from 'components/custom-link/custom-link.component'

const props = {
  to: '/test-to',
  onClick: jest.fn(),
  updateStore: jest.fn(),
  className: 'test-class',
}

let wrapper
describe('components - CustomLink', () => {
  beforeEach(() => {
    wrapper = shallow(<CustomLink {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render children provided', () => {
    const Child = () => <div>child</div>
    wrapper.setProps({ children: <Child /> })
    expect(wrapper.find(Child).length).toBe(1)
  })

  it('should update the store with the `to` on click', () => {
    wrapper.simulate('click')
    expect(props.updateStore).toBeCalledWith(props.to)
  })

  it('should call the `onClick` passed on click', () => {
    wrapper.simulate('click')
    expect(props.onClick).toBeCalled()
  })

  it('should not throw on click if no `onClick` passed', () => {
    wrapper.setProps({ onClick: undefined })

    let error
    try {
      wrapper.simulate('click')
    } catch (err) {
      error = err
    } finally {
      expect(error).toBeUndefined()
    }
  })
})