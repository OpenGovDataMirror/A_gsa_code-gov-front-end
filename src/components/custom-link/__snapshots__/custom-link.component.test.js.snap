// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - CustomLink should render correctly 1`] = `
<Link
  className="test-class"
  onClick={[Function]}
  replace={false}
  to="/test-to"
/>
`
